internal/app/machined/pkg/runtime/v1alpha1/platform/openstack/openstack.go

```diff
@@ -34,82 +32,71 @@ const (
 type Openstack struct{}
 
 // Name implements the runtime.Platform interface.
-func (a *Openstack) Name() string {
+func (o *Openstack) Name() string {
 	return "openstack"
 }
 
 // Configuration implements the runtime.Platform interface.
-func (a *Openstack) Configuration(ctx context.Context) ([]byte, error) {
+func (o *Openstack) Configuration(ctx context.Context) ([]byte, error) {
 	log.Printf("fetching machine config from: %q", OpenstackUserDataEndpoint)
 
-	return download.Download(ctx, OpenstackUserDataEndpoint,
+	machineConfigDl, err := download.Download(ctx, OpenstackUserDataEndpoint,
 		download.WithErrorOnNotFound(errors.ErrNoConfigSource),
 		download.WithErrorOnEmptyResponse(errors.ErrNoConfigSource))
+	if err != nil {
+		return nil, err
+	}
+
+	// Some openstack setups does not allow you to change user-data,
+	// so skip this case.
+	if bytes.HasPrefix(machineConfigDl, []byte("#cloud-config")) {
+		return nil, errors.ErrNoConfigSource
+	}
+
+	return machineConfigDl, nil
 }
 
 // Mode implements the runtime.Platform interface.
-func (a *Openstack) Mode() runtime.Mode {
+func (o *Openstack) Mode() runtime.Mode {
 	return runtime.ModeCloud
 }
 
 // Hostname implements the runtime.Platform interface.
-func (a *Openstack) Hostname(ctx context.Context) (hostname []byte, err error) {
-	req, err := http.NewRequestWithContext(ctx, http.MethodGet, OpenstackHostnameEndpoint, nil)
-	if err != nil {
-		return nil, err
-	}
+func (o *Openstack) Hostname(ctx context.Context) (hostname []byte, err error) {
+	log.Printf("fetching hostname from: %q", OpenstackHostnameEndpoint)
 
-	resp, err := http.DefaultClient.Do(req)
+	hostname, err = download.Download(ctx, OpenstackHostnameEndpoint,
+		download.WithErrorOnNotFound(errors.ErrNoHostname),
+		download.WithErrorOnEmptyResponse(errors.ErrNoHostname))
 	if err != nil {
-		return nil, err
+		// Platform cannot support this endpoint, or return timeout.
+		// ApplyDynamicConfig can crush in this situation.
+		return hostname, nil //nolint:nilerr
```

Comment:
I don't think completely ignoring error is the right way here, as if someone sets the hostname, and it doesn't get set on Talos side, that would be hard to debug

if we have to ignore errors (why?) we should at least log the failure

Reply:
I've got this case, after apply machineconfig, network was changed. And ApplyDynamicConfig cannot get the hostname because "no route to host" issue.

and here https://github.com/talos-systems/talos/blob/7c5045bd929fcf5028cae3840970e692ef3bc7c9/pkg/machinery/config/types/v1alpha1/v1alpha1_provider.go#L88-L94

ApplyDynamicConfig catch the nil result...